src/utils/structs/storage_array.cairo

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.7.0 (utils/structs/storage_array.cairo)
+
+use array::ArrayTrait;
+use poseidon::poseidon_hash_span;
+use starknet::{
+    StorageBaseAddress, Store, SyscallResultTrait, SyscallResult, storage_address_from_base,
+    storage_base_address_from_felt252, storage_read_syscall, storage_write_syscall
+};
+
+const NOT_IMPLEMENTED: felt252 = 'Not implemented';
+
+/// Represents an Array that can be stored in storage.
+#[derive(Copy, Drop)]
+struct StorageArray<T> {
+    address_domain: u32,
+    base: StorageBaseAddress
+}
+
+impl StoreStorageArray<T, impl TDrop: Drop<T>, impl TStore: Store<T>> of Store<StorageArray<T>> {
+    #[inline(always)]
+    fn read(address_domain: u32, base: StorageBaseAddress) -> SyscallResult<StorageArray<T>> {
+        SyscallResult::Ok(StorageArray { address_domain, base })
+    }
+    #[inline(always)]
+    fn write(
+        address_domain: u32, base: StorageBaseAddress, value: StorageArray<T>
+    ) -> SyscallResult<()> {
+        SyscallResult::Err(array![NOT_IMPLEMENTED])
+    }
+    #[inline(always)]
+    fn read_at_offset(
+        address_domain: u32, base: StorageBaseAddress, offset: u8
+    ) -> SyscallResult<StorageArray<T>> {
+        SyscallResult::Err(array![NOT_IMPLEMENTED])
+    }
+    #[inline(always)]
+    fn write_at_offset(
+        address_domain: u32, base: StorageBaseAddress, offset: u8, value: StorageArray<T>
+    ) -> SyscallResult<()> {
+        SyscallResult::Err(array![NOT_IMPLEMENTED])
+    }
+    #[inline(always)]
+    fn size() -> u8 {
+        // 0 was selected because the read method doesn't actually read from storage.
+        0_u8
+    }
+}
+
+/// Trait for accessing a storage array.
+///
+/// `read_at` and `write_at` don't check the len of the array, caution must be exercised.
+/// The current len of the array is stored at the base StorageBaseAddress as felt.
```

Comment:
In the way we are using the module overflows will never happen, so that would be an unnecessary overhead in steps and gas cost. The comment is for potential users trying to use the module directly themselves.